### ISSUE TITLE
Creation of Azure VM fails while using an Existing Storage account 

### ISSUE THREAD

**Issue by k2velocity** [Azure/azure-xplat-cli#2852]
CLI Version: **0.9.X**
OS Type:  Win / Linux (Centos) 
Installation via:  

Mode: ARM

Description:

While creating a Virtual machine from an existing OS disk and Storage account with the -o switch the command first complains that the Storage account did not exists and then tried creating it and then it  fails with Storage account already exists 

PS C:\WINDOWS\system32> azure vm create Rresourcename VMname eastus2 -y Windows -o storageaccountname -d "https://quanta0015938.blob.core.windows.net/vhds/GN024.vhd" -z Standard_A1 -F “vnetname” -j “subnetname” -f “Vmname_nic”
info:    Executing command vm create
+ Looking up the VM "Vmname"
info:    Using the VM Size "Standard_A1"
+ Looking up the NIC "Vmname_nic"
info:    An nic with given name "Vmname_nic" not found, creating a new one
+ Looking up the virtual network "Vmname"
info:    Found an existing virtual network "Vmname"
info:    Existing Subnets:
info:      default:10.10.0.0/24
info:      Backend:10.10.1.0/24
info:    Verifying subnet
+ Looking up the subnet "default" under the virtual network "Vnetname"
info:    Subnet with given name "default" exists under the virtual network "Vnetname", using this subnet
info:    No public ip parameters found, the ip configuration of new NIC will have only subnet configured
+ Creating NIC "Vmname_nic"
+ Looking up the NIC "Vmname_nic"
+ Looking up the storage account storageaccountname
info:    Could not find the storage account "storageaccountname", trying to create new one
+ Creating storage account "storageaccountname" in "eastus2"
error:   The storage account named storageaccountname already exists under the subscription.
info:    Error information has been recorded to C:\Users\user\.azure\azure.err
error:   vm create command failed

**Comment by amarzavery** [comment]
@sauryadas @huangpf - Please take a look at this issue

**Comment by huangpf** [comment]
It seems the existing storage account is in another location which is not ```eastus2```. @k2velocity Could you please help confirm?

Could you please try the following: 1) create the VM in the same region as the existing storage account's, 2) refer to a new storage account that is in ```eastus2```, and 3) omit the ```-o``` input.

**Comment by k2velocity** [comment]
The VM that i am creating is from an existing OS disk so the VHD path  is from East US 2 location , also this is a standard Storage so wanted to Use same storage account for Diagnostics logging as well


If i Skip the -o Flag then the command complete by

1. Creating a new Storage account in East US 2 for the diagnostics
2. Creates the VM with the Existing data disk in East US 2

**Comment by huangpf** [comment]
OK. Will look into it. But just want to know that you are unblocked for now by omitting the ```-o``` parameter.

@sauryadas Is storage account's charge by number of accounts, or just storage amount? If it's the latter, I think it's at least not charging @k2velocity more by the current setup.

**Comment by k2velocity** [comment]
@huang: are you  looking out for more details from the command ?

**Comment by k2velocity** [comment]
Any updates on this ?

**Comment by huangpf** [comment]
+@shraddhaagrawal to help take a look.

**Comment by shraddhaagrawal** [comment]
@k2velocity 
It seems like existing storage account is available under your subscription but not under the mentioned resource group.
Could you please check?
Please try creating vm in same resource group as that of existing storage account.

**Comment by k2velocity** [comment]
@sharadagarwal : The Storage account is part of the same resource group as we can create the Vm if we skipped the diagnostics

**Comment by shraddhaagrawal** [comment]
@k2velocity I have successfully created vm using existing storage account. command will throw mentioned error If storage account is not part of the same resource group.

# Description-

### Same Resource Group
`$azure vm create testrg testvm westus -y Windows -o storageaccountname -d https://storageaccountname.blob.core.windows.net/images/testvm2016530142717.vhd -z Basic_A0 -F vnet2852 -j subnet2852 -f testnic2852`

logs-
```
info:    Executing command vm create
info:    Looking up the VM "testvm"
info:    Using the VM Size "Basic_A0"
info:    Looking up the NIC "testnic2852"
info:    An nic with given name "testnic2852" not found, creating a new one
info:    Looking up the virtual network "vnet2852"
info:    Found an existing virtual network "vnet2852"
info:    Existing Subnets:
info:      subnet2852:10.0.0.0/24
info:    Verifying subnet
info:    Looking up the subnet "subnet2852" under the virtual network "vnet2852"
info:    Subnet with given name "subnet2852" exists under the virtual network "vnet2852", using this subnet
info:    No public ip parameters found, the ip configuration of new NIC will have only subnet configured
info:    Creating NIC "testnic2852"
info:    Looking up the NIC "testnic2852"
info:    Looking up the storage account storageaccountname
info:    The storage URI 'https://storageaccountname.blob.core.windows.net/' will be used for boot diagnostics settings, and it can be overwritten by the parameter input of '--boot-diagnostics-storage-uri'.
info:    Getting virtual machine and/or extension image publishers (Location: "westus")
info:    Getting virtual machine extension image types (Publisher: "Microsoft.Compute" Location:"westus")
info:    Getting virtual machine extension image verions (Publisher: "Microsoft.Compute" Type:"BGInfo" Location:"westus")
info:    Creating VM "testvm"
info:    Installing extension "BGInfo", VM: "testvm"
info:    vm create command OK

```

### Different Resource Group
`$azure vm create testrgnew testvm westus -y Windows -o storageaccountname -d https://storageaccountname.blob.core.windows.net/images/testvm2016530142717.vhd -z Basic_A0 -F vnet2852 -j subnet2852 -f testnic2852`

logs-
```
info:    Executing command vm create
+ Looking up the VM "testvm"
info:    Using the VM Size "Basic_A0"
+ Looking up the NIC "testnic2852"
info:    An nic with given name "testnic2852" not found, creating a new one
+ Looking up the virtual network "vnet2852"
info:    Preparing to create new virtual network and subnet
Enter vnet address prefix: 10.0.0.0/16
Enter vnet subnet address prefix: 10.0.0.0/24
\ Creating a new virtual network "vnet2852" [address prefix: "10.0.0.0/16"] with
+subnet "subnet2852" [address prefix: "10.0.0.0/24"]
+ Looking up the virtual network "vnet2852"
+ Looking up the subnet "subnet2852" under the virtual network "vnet2852"
info:    No public ip parameters found, the ip configuration of new NIC will have only subnet configured
+ Creating NIC "testnic2852"
+ Looking up the NIC "testnic2852"
+ Looking up the storage account storageaccountname
info:    Could not find the storage account "storageaccountname", trying to create new one
+ Creating storage account "storageaccountname" in "westus"
error:   The storage account named storageaccountname already exists under the subscription.
info:    Error information has been recorded to C:\Users\user\.azure\azure.err
error:   vm create command failed
```

Please correct me if i am missing something.
